src/Host/Client/Impl/rtvs/R/overrides.R

```diff
@@ -54,6 +54,20 @@ show_file <- function(files, header, title, delete.file) {
     }
 }
 
+
+# Override for edit(...). Opens file in VS editor and blocks until the file is closed.
+edit_file <- function(name, file, title) {
+    if(!is.null(name) && is.function(name)) {
+        source <- send_request_and_get_response("?EditFile", paste0(deparse(name), collapse = '\n'), NULL);
+        parsed <- try(eval.parent(parse(text = source)));
+        return(parsed);
+    } else if(!is.null(file)) {
+        send_request_and_get_response("?EditFile", NULL, file);
+    } else {
+        utils:::edit(name, file, title);
```

Comment:
That's because when it's a number, it doesn't even hit our `editor` function. That's what I meant when I said that the third branch is a it-never-happens case. We shouldn't be trying to call `utils::edit` in it - if it *does* actually ever get invoked, that would be wrong (recursion), and if it doesn't, then that code doesn't do anything useful (but implies that it does).